cpu/msp430/mtarch.c

```diff
@@ -66,10 +66,10 @@ mtarch_start(struct mtarch_thread *t,
 
   t->sp = &t->stack[MTARCH_STACKSIZE - 1];
 
-  *t->sp = (unsigned short)mt_exit;
+  *t->sp = (unsigned short)(void *)mt_exit;
```

Comment:
My suggestion then is to revert this particular fix to leave the warning in there, as long as we don't have a better fix (actually fixing or removing mtarch in the 20-bit case)